src/graphics/d3d12.rs

```diff
@@ -0,0 +1,383 @@
+use std::ffi::{c_void, CString};
+use std::sync::atomic::AtomicBool;
+use std::sync::{Arc, Mutex};
+
+use anyhow::Context;
+use bevy::math::uvec2;
+use bevy::prelude::*;
+use bevy::render::renderer::{RenderAdapter, RenderAdapterInfo, RenderDevice, RenderQueue};
+use bevy::window::RawHandleWrapper;
+use openxr as xr;
+use wgpu::Instance;
+use wgpu_hal::{Adapter as HalAdapter, Instance as HalInstance};
+use winapi::shared::dxgiformat::{self, DXGI_FORMAT};
+use winapi::{shared::dxgi, um::d3dcommon};
+
+use crate::input::XrInput;
+use crate::resources::{
+    Swapchain, SwapchainInner, XrEnvironmentBlendMode, XrFormat, XrFrameState, XrFrameWaiter,
+    XrInstance, XrResolution, XrSession, XrSessionRunning, XrSwapchain, XrViews,
+};
+use crate::VIEW_TYPE;
+
+pub fn initialize_xr_graphics(
+    window: Option<RawHandleWrapper>,
+    xr_entry: xr::Entry,
+    available_extensions: xr::ExtensionSet,
+    // Horrible hack to get the Handtacking extension Loaded, Replace with good system to load
+    // any extension at some point
+) -> anyhow::Result<(
+    RenderDevice,
+    RenderQueue,
+    RenderAdapterInfo,
+    RenderAdapter,
+    Instance,
+    XrInstance,
+    XrSession,
+    XrEnvironmentBlendMode,
+    XrResolution,
+    XrFormat,
+    XrSessionRunning,
+    XrFrameWaiter,
+    XrSwapchain,
+    XrInput,
+    XrViews,
+    XrFrameState,
+    // Horrible hack to get the Handtacking extension Loaded, Replace with good system to load
```

Comment:
So IDK what this comment means, it's copied from the Vulkan backend. I preserved it because presumably it's a note that needs addressing down the line.